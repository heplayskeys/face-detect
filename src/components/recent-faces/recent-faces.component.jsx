import React from 'react';
import { connect } from 'react-redux';
import { setInputField } from '../../redux/input/input.actions';
import './recent-faces.styles.scss';

export const RecentFaces = ({ recentGrabs, route, setInput }) => {
	const grabs = new Set();

	for (let i = recentGrabs.length - 1; i >= 0; i--) {
		if ([...grabs].length === 4) {
			break;
		}

		if (grabs.has(recentGrabs[i])) {
			continue;
		} else {
			grabs.add(recentGrabs[i]);
		}
	}

	const handleClick = e => {
		const src = e.target.src;
		setInput(src);
	};

	return route === 'home' ? (
		<div className='recent-grabs-container'>
			<h3 className='tr'>{'Recent Grabs:'}</h3>
			{[...grabs].map((grab, idx) => (
				<img
					className='recent-grab grower dimmer db mw5 br2 ba b--black-10 shadow-1 pointer'
					src={grab}
					alt='recent-grab'
					key={idx}
					onClick={handleClick}
				/>
			))}
		</div>
	) : null;
};

const mapStateToProps = state => ({
	input: state.setInput.input,
	route: state.setRoute.route
});

const mapDispatchToProps = dispatch => ({
	setInput: input => dispatch(setInputField(input))
});

export default connect(mapStateToProps, mapDispatchToProps)(RecentFaces);
