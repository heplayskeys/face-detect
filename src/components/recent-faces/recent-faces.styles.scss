.recent-grabs-container {
	width: 100%;
	position: absolute;
	bottom: 100px;
	display: grid;
	grid-template-columns: repeat(5, auto);

	.recent-grab {
		margin: 0 auto;
		width: 100px;
		height: 100px;
		object-fit: cover;
		opacity: 0.5;
		transition: all 0.3s;

		&:hover {
			opacity: 1;
			transform: scale(1.2);
		}
	}
}
