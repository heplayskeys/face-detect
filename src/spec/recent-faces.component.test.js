import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { RecentFaces } from '../components/recent-faces/recent-faces.component';

let wrapper;

describe('<RecentFaces />', () => {
	beforeAll(() => {
		const mockProps = {
			recentGrabs: [0],
			route: 'home',
			setInput: jest.fn()
		};
		wrapper = shallow(<RecentFaces {...mockProps} />);
	});

	it('renders the recent faces component with length 1', () => {
		expect(shallowToJson(wrapper)).toMatchSnapshot();
		expect(wrapper.find('img')).toHaveLength(1);
	});

	it('renders the recent faces component with length 3', () => {
		wrapper.setProps({ recentGrabs: [0, 1, 2] });
		expect(shallowToJson(wrapper)).toMatchSnapshot();
		expect(wrapper.find('img')).toHaveLength(3);
	});

	it('renders the recent faces component with length 4', () => {
		wrapper.setProps({ recentGrabs: [0, 1, 2, 3] });
		expect(shallowToJson(wrapper)).toMatchSnapshot();
		expect(wrapper.find('img')).toHaveLength(4);
	});

	it('renders the recent faces component with length 4 as maximum count', () => {
		wrapper.setProps({ recentGrabs: [0, 1, 2, 3, 4, 5, 6, 7] });
		expect(shallowToJson(wrapper)).toMatchSnapshot();
		expect(wrapper.find('img')).toHaveLength(4);
	});
});
