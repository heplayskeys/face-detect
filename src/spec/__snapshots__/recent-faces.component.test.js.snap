// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecentFaces /> renders the recent faces component with length 1 1`] = `
<div
  className="recent-grabs-container"
>
  <h3
    className="tr"
  >
    Recent Grabs:
  </h3>
  <img
    alt="recent-grab"
    className="recent-grab grower dimmer db mw5 br2 ba b--black-10 shadow-1 pointer"
    key="0"
    onClick={[Function]}
    src={0}
  />
</div>
`;

exports[`<RecentFaces /> renders the recent faces component with length 3 1`] = `
<div
  className="recent-grabs-container"
>
  <h3
    className="tr"
  >
    Recent Grabs:
  </h3>
  <img
    alt="recent-grab"
    className="recent-grab grower dimmer db mw5 br2 ba b--black-10 shadow-1 pointer"
    key="0"
    onClick={[Function]}
    src={2}
  />
  <img
    alt="recent-grab"
    className="recent-grab grower dimmer db mw5 br2 ba b--black-10 shadow-1 pointer"
    key="1"
    onClick={[Function]}
    src={1}
  />
  <img
    alt="recent-grab"
    className="recent-grab grower dimmer db mw5 br2 ba b--black-10 shadow-1 pointer"
    key="2"
    onClick={[Function]}
    src={0}
  />
</div>
`;

exports[`<RecentFaces /> renders the recent faces component with length 4 1`] = `
<div
  className="recent-grabs-container"
>
  <h3
    className="tr"
  >
    Recent Grabs:
  </h3>
  <img
    alt="recent-grab"
    className="recent-grab grower dimmer db mw5 br2 ba b--black-10 shadow-1 pointer"
    key="0"
    onClick={[Function]}
    src={3}
  />
  <img
    alt="recent-grab"
    className="recent-grab grower dimmer db mw5 br2 ba b--black-10 shadow-1 pointer"
    key="1"
    onClick={[Function]}
    src={2}
  />
  <img
    alt="recent-grab"
    className="recent-grab grower dimmer db mw5 br2 ba b--black-10 shadow-1 pointer"
    key="2"
    onClick={[Function]}
    src={1}
  />
  <img
    alt="recent-grab"
    className="recent-grab grower dimmer db mw5 br2 ba b--black-10 shadow-1 pointer"
    key="3"
    onClick={[Function]}
    src={0}
  />
</div>
`;

exports[`<RecentFaces /> renders the recent faces component with length 4 as maximum count 1`] = `
<div
  className="recent-grabs-container"
>
  <h3
    className="tr"
  >
    Recent Grabs:
  </h3>
  <img
    alt="recent-grab"
    className="recent-grab grower dimmer db mw5 br2 ba b--black-10 shadow-1 pointer"
    key="0"
    onClick={[Function]}
    src={7}
  />
  <img
    alt="recent-grab"
    className="recent-grab grower dimmer db mw5 br2 ba b--black-10 shadow-1 pointer"
    key="1"
    onClick={[Function]}
    src={6}
  />
  <img
    alt="recent-grab"
    className="recent-grab grower dimmer db mw5 br2 ba b--black-10 shadow-1 pointer"
    key="2"
    onClick={[Function]}
    src={5}
  />
  <img
    alt="recent-grab"
    className="recent-grab grower dimmer db mw5 br2 ba b--black-10 shadow-1 pointer"
    key="3"
    onClick={[Function]}
    src={4}
  />
</div>
`;
